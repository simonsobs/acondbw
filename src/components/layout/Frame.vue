<template>
  <router-view name="navigationDrawer" v-model="drawer"></router-view>
  <router-view name="appBar" v-slot="{ Component }">
    <component :is="Component" :order="order">
      <template v-slot:prepend>
        <v-app-bar-nav-icon @click="toggleDrawer" v-if="mobile">
        </v-app-bar-nav-icon>
      </template>
    </component>
  </router-view>
</template>

<script setup lang="ts">
import { ref, computed, watchEffect } from "vue";
import { useDisplay } from "vuetify";

// https://vuetifyjs.com/en/features/display-and-platform/
const { mobile } = useDisplay();

const drawer = ref<boolean>(false);
watchEffect(() => {
  drawer.value = !mobile.value;
});

function toggleDrawer() {
  drawer.value = !drawer.value;
}

// https://vuetifyjs.com/en/features/application-layout/#dynamic-layouts-and-order
const order = computed(() => (mobile.value ? 0 : -1));
</script>
