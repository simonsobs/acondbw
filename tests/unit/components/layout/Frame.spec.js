import Vue from "vue";
import Vuetify from "vuetify";
import { mount, shallowMount, createLocalVue } from "@vue/test-utils";

import Frame from "@/components/layout/Frame";

Vue.use(Vuetify);

describe("Frame.vue", () => {
  let localVue;
  let vuetify;
  let wrapper;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = shallowMount(Frame, {
      localVue,
      vuetify,
    });
  });

  it("match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
