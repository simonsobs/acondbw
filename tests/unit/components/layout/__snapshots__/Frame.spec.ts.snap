// Vitest Snapshot v1

exports[`Frame.vue > match snapshot 1`] = `
"<div>
  <vuecomponent-stub app=\\"true\\" mobilebreakpoint=\\"1264\\" clipped=\\"true\\" height=\\"100vh\\" minivariantwidth=\\"56\\" src=\\"\\" tag=\\"nav\\" width=\\"256\\">
    <anonymous-stub></anonymous-stub>
  </vuecomponent-stub>
  <anonymous-stub></anonymous-stub>
</div>"
`;
